### PR TITLE
Support placeholder Garbage Collection frames in Collectors::Stack

### PR DESCRIPTION
**Pull request by ivoanjo**
**What does this PR do?**:

This PR adds a new feature to the `Collectors::Stack`: being able to include, at the top of the sampled stack, a placeholder frame saying "Garbage Collection".

**Motivation**:

The "Garbage Collection" placeholder is part of a new feature to show on the flamegraph the CPU and wall-clock time spent in garbage collection.

Right now we are not yet collecting this information, but I will follow up with a later PR to add this feature to `Collectors::CpuAndWallTime`.

**Additional Notes**:

~~I'm opening this PR as a draft because:~~
* ~~it's on top of #2302 to avoid a later rebase~~
* ~~libdatadog 0.9 has not been released yet, so CI fails~~

~~Thus, once libdatadog 0.9 is released and #2302 is merged, I'll re-trigger CI and mark this PR as non-draft when CI is back to green. I don't expect to need any code changes for that.~~

**How to test the change?**:

The changes to `Collectors::Stack` include code coverage. I plan to add later integration tests for the whole "gather GC time" feature.

### CODE REVIEW

**Comment by ivoanjo** [comment]
Update: CI is now green for this change. #2302 has not been merged yet (so this PR is still on top of it) BUT it won't change anything for this PR, so I'm marking it as ready for review.

File: ext/ddtrace_profiling_native_extension/collectors_stack.c [hunk 1]
```diff
@@ -80,20 +112,88 @@ static VALUE _native_sample(DDTRACE_UNUSED VALUE _self, VALUE thread, VALUE reco
...
   sampling_buffer *buffer = sampling_buffer_new(max_frames_requested);
 
-  sample_thread(
-    thread,
-    buffer,
-    recorder_instance,
-    (ddog_Slice_i64) {.ptr = metric_values, .len = ENABLED_VALUE_TYPES_COUNT},
-    (ddog_Slice_label) {.ptr = labels, .len = labels_count}
-  );
+  if (in_gc == Qnil || in_gc == Qfalse) {
```
Comment (marcotc) [inline ext/ddtrace_profiling_native_extension/collectors_stack.c:115]:
```suggestion
  if (!RB_TEST(in_gc)) {
```
https://github.com/ruby/ruby/blob/7837dccb0ea9f84f033635e77bc9dfb6ea023bcd/include/ruby/internal/special_consts.h#L123-L149

**Review by marcotc — APPROVED** [review]
I feel very enterprisy reviewing all this C code.

File: ext/ddtrace_profiling_native_extension/collectors_stack.c (hunk 1 above)
Comment (ivoanjo) [reply ext/ddtrace_profiling_native_extension/collectors_stack.c]: I'll admit I thought "wait I think there was a better way of doing it..." and I did search a bit but I didn't find it. Thanks a lot for the reminder! :)

**Comment by ivoanjo** [comment]
> I feel very enterprisy reviewing all this C code.

In my defense, at least it has some beautiful RSpec BDD specs to clean your eyes with ;)

**Pull request merged**